### PR TITLE
Prod fix

### PR DESCRIPTION
**Pull request by fbertsch**
This limits to recent FX versions, and also fixes the recent difference of processes.

### CODE REVIEW

**Comment by fbertsch** [comment]
Fixes #62.

File: mozilla_schema_generator/main_ping.py [hunk 1]
```diff
@@ -14,6 +14,11 @@
 
 class MainPing(GenericPing):
 
+    # Only includes probes that have been available at some point past
+    # this version.
+    # ONLY DECREMENT, or the schema will change in an incompatible way!
+    MIN_FX_VERSION = 60
```
Comment (acmiyaguchi) [inline mozilla_schema_generator/main_ping.py]: Any particular reason to choose Firefox 60?

Comment (fbertsch) [reply mozilla_schema_generator/main_ping.py]: I chose this because that is what ESR is on right now. Though we are pushing to 68 very soon.

**Comment by acmiyaguchi** [comment]
A preliminary diff between 4dc43bd759ceb809ad1d0f706065903fabaf2769 and 463375f566c39fb4e2f88ca1e1bfcb85ba6b66eb shows that there are more deletions than additions. It's not obvious that this patch actually fixes the schema to do the right thing.

I created a diff on a fresh `test-generated-schemas` branch: https://github.com/mozilla-services/mozilla-pipeline-schemas/commit/37eb784bba5cfd2d79dc994489c2bd1b4e61ace5

```
# delete the branch

# generate the initial commit into `test-generated-schemas`
export MPS_SSH_KEY_BASE64=$(cat ~/.ssh/id_rsa | base64)
git checkout 4dc43bd759ceb809ad1d0f706065903fabaf2769
docker-compose run --entrypoint bash app
mozilla-schema-generator/bin/schema-generator.sh
exit

# generate the follow-up commit
git checkout prod_fix
make build && make run
```

From the generated schemas branch, it may be useful to run the following command:

```
git diff HEAD~2 HEAD -- schemas/telemetry/main/main.4.bq 
```

I'm going to test that this schema works properly in BQ, otherwise spend some time trying to update this patch.

**Comment by acmiyaguchi** [comment]
```
# assumption: test-generated-schemas contains two consecutive commits, the first 
# containing schema from MSG@4dc43bd followed by MSG@463375f, which is also HEAD
cd mozilla-pipeline-schemas
git checkout test-generated-schemas
git checkout HEAD~1
bq mk test_schemas
bq mk --table --schema schemas/telemetry/main/main.4.bq --time_partitioning_field submission_timestamp test_schemas.main_v4

git checkout test-generated-schemas
bq update test_schemas.main_v4 schemas/telemetry/main/main.4.bq
```

This leads to the following error:

```
BigQuery error in update operation: Provided Schema does not match Table amiyaguchi-dev:test_schemas.main_v4. Field payload.histograms.application_reputation_allowlist_match is missing in
new schema
```

**Comment by acmiyaguchi** [comment]
Based on some further discussion, I've verified that this patch is doing the correct thing by generating another diff with the following patch:

```diff
diff --git a/mozilla_schema_generator/main_ping.py b/mozilla_schema_generator/main_ping.py
index 4ec2077..7f2b3b2 100644
--- a/mozilla_schema_generator/main_ping.py
+++ b/mozilla_schema_generator/main_ping.py
@@ -17,7 +17,7 @@ class MainPing(GenericPing):
     # Only includes probes that have been available at some point past
     # this version.
     # ONLY DECREMENT, or the schema will change in an incompatible way!
-    MIN_FX_VERSION = 60
+    MIN_FX_VERSION = 30
 
     schema_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/master/schemas/telemetry/main/main.4.schema.json" # noqa E501
     env_url = "https://raw.githubusercontent.com/mozilla-services/mozilla-pipeline-schemas/master/templates/include/telemetry/environment.1.schema.json" # noqa E501
````

I've pushed this to the existing `test-generated-schemas` branch. The following commands are done locally after checking out the branch.

`HEAD` corresponds to the applied patch (`MIN_FX_VERSION = 30`)
`HEAD~1` corresponds to the current PR (`MIN_FX_VERSION = 60`)
`HEAD~2` corresponds to 4dc43bd

```
$ git diff --stat HEAD~2 HEAD~1 -- schemas/telemetry/main/main.4.bq
 schemas/telemetry/main/main.4.bq | 9834 +++++++++++++++++++++++++++++++++++++++++++++++++++---------------------------------------------------------------------------
 1 file changed, 3995 insertions(+), 5839 deletions(-)
$ git diff --stat HEAD~2 HEAD -- schemas/telemetry/main/main.4.bq
 schemas/telemetry/main/main.4.bq | 525 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 525 insertions(+)
```

Running `bq update test_schemas.main_v4 schemas/telemetry/main/main.4.bq` succeeds.

**Comment by whd** [comment]
Generally I don't like the idea of dropping and recreating the main ping table with the day of the production deploy. It does not lend confidence that our pipeline and schemas are stable.

We're planning on backfilling a year's worth of data into these tables. As long as we're clear that this new schema has been "certified" as the last time we need to drop and recreate tables, I'm ok with moving forward with this.

**Comment by acmiyaguchi** [comment]
@fbertsch There's plenty of space before we hit the column limit, so I backed the minimum version down to 30. Having the last month or so of data in the non-prod table is nice, while we figure out a way to backfill the prod tables.

**Pull request merged**